src/components/email/VerifyEmail.tsx

```diff
@@ -0,0 +1,139 @@
+import { useRouter } from 'next/router';
+import { useContext, useEffect, useState } from 'react';
+import UserContext from '../../modules/BuilderPlace/context/UserContext';
+import { useMutation } from 'react-query';
+import { IVerifyEmail } from '../../app/api/emails/verify/route';
+import axios, { AxiosResponse } from 'axios';
+import Loading from '../Loading';
+import {
+  EMAIL_ALREADY_VERIFIED,
+  EMAIL_VERIFIED_SUCCESSFULLY,
+} from '../../modules/BuilderPlace/apiResponses';
+
+const verifyEmail = () => {
+  const router = useRouter();
+  const { id } = router.query;
+  const { getUser } = useContext(UserContext);
+  const [loading, setLoading] = useState(true);
+  const [pageResponse, setPageResponse] = useState('Missing Id');
+  const [redirect, setRedirect] = useState(true);
+  const emailMutation = useMutation(
+    async (body: IVerifyEmail): Promise<AxiosResponse<{ id: string; message: string }>> => {
+      return await axios.post('/api/emails/verify', body);
+    },
+  );
+
+  useEffect(() => {
```

Comment:
here you don't need a use effect, it's better for perf to prevent a rerender with just: 

```
const domain = window.location.hostname + ':' + window.location.port;
const redirect = domain === process.env.NEXT_PUBLIC_ROOT_DOMAIN;
```

- Also it's important to respect boolean naming convention (start with is, has...)
- and the name "redirect" is not really clear, impossible to understand without checking the code. It could be isOnRootDomain